lib/chef/provider.rb

```diff
@@ -286,10 +286,27 @@ def self.include_resource_dsl_module(resource)
           resource.class.properties.each do |name, property|
             class_eval(<<-EOM, __FILE__, __LINE__)
               def #{name}(*args, &block)
+                # not making this a formal deprecation yet because I don't want to drive tim completely crazy, this
+                # should be upgraded to a real deprecation in 13.1 or so
+                Chef::Log.warn "Deprecated property access: please replace calling #{name} with new_resource.#{name}"
```

Comment:
Thinking more on this, I'm not sure I love this. Deprecation warnings are automatically de-dup'd to avoid spamming the logs, but this won't be.